fix(bhDateInterval): ensure date change is called

jniles opened the pull request:
This commit fixes a date change bug in the bhDateInterval that prevented updates to the minimum limit of the date.

Fixes #3628.

jniles committed:
fix(bhDateInterval): ensure date change is called

This commit fixes a date change bug in the bhDateInterval that prevented
updates to the minimum limit of the date.

Fixes #3628.

mbayopanda commented on client/src/js/components/bhDateInterval.js:
```diff
@@ -62,20 +62,32 @@ function bhDateInterval(moment, bhConstants, Fiscal, Session) {
   function search(selection) {
     $ctrl.selected = selection.translateKey;
     selection.fn();
+    $ctrl.onChangeDate();
```
Good 👍

mbayopanda commented on client/src/js/components/bhDateInterval.js:
```diff
@@ -62,20 +62,32 @@ function bhDateInterval(moment, bhConstants, Fiscal, Session) {
+    if ($ctrl.onChange) {
+      $ctrl.onChange();
+    }
+  };
```
Nice 👍

mbayopanda commented on client/src/js/components/bhDateInterval.js:
```diff
@@ -62,20 +62,32 @@ function bhDateInterval(moment, bhConstants, Fiscal, Session) {
+      getMinimumFiscalYearDate();
     }
 
     $ctrl.pickerToOptions = { showWeeks : false, minDate : $ctrl.dateFrom };
```
I think this line must be before the ```if (angular.isDefined($ctrl.limitMinFiscal)) {```

jniles replied:
Good point!

mbayopanda reviewed (approved):
The code LGTM and works fine

I just noticed that in the fiscal year module, we have to click twice on date key (today, this week, this month, this year) for seeing change for `Nombre de mois`

jniles committed:
fix(fiscal): use updated date for num months calc

The fiscal year number of months calculation was using outdated data for
the calculation, since the updates were not propagated before the
callback was firing.  Now, we make sure that the calculation uses dates
from the bhDateInterval controller directly.
